lib/rubocop/cop/rails/active_record_override.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Checks for overriding built-in Active Record methods instead of using
+      # callbacks.
+      #
+      # @example
+      #   # bad
+      #   class X < ApplicationRecord
+      #     def save
+      #       self.title = title.upcase!
+      #       super
+      #     end
+      #   end
+      #
+      #   # good
+      #   class X < ApplicationRecord
+      #     before_save upcase_title
+      #
+      #     def upcase_title
+      #       self.title = title.upcase!
+      #     end
+      #   end
+      #
+      class ActiveRecordOverride < Cop
+        MSG = 'Use Active Record callbacks instead of overriding methods.'
```

Comment:
This message could be a bit more specific - Use X callback instead of overriding Y method.